src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1857,40 +1857,79 @@ protected String getArchiveEmail(String sessionId, String siteId) {
 		}
 
 		// Breakout the email name used in CTools for this archive.
-		protected String extractArchiveEmailName(String emailArchive) {
+		static public String extractArchiveEmailName(String emailArchive) {
+			String archive_email_name = null;
 			JSONObject jo = new JSONObject(emailArchive);
 			JSONArray ja = jo.getJSONArray("mailarchive_collection");
-			JSONArray firstHeaders = (JSONArray) ((JSONObject)ja.get(0)).get("headers");
-
-			int myJsonArraySize = firstHeaders.length();
-			String archive_email_name = null;
-
-			for (int i = 0; i < myJsonArraySize && archive_email_name == null; i++) {
-				String header  = (String) firstHeaders.get(i);
-				// look for the To: header
-				if (! header.startsWith("To: ")) {
-					continue;
+			outerloop:
+			for(int j=0;j<ja.length();j++) {
```

Comment:
For clarity document that this loop will make sure that archive_email_name is set to one of:
- a valid group name based on the first To: header it finds 
- null

That's worth doing as it can take a few minutes to figure out what a double loop with a continue and two different breaks actually does.